onnxruntime/core/optimizer/isinf_reducesum_fusion.cc

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/isinf_reducesum_fusion.h"
+
+#include "onnx/defs/attr_proto_util.h"
+
+#include "core/optimizer/initializer.h"
+#include "core/graph/graph_utils.h"
+#include "core/framework/tensorprotoutils.h"
+#include <deque>
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+namespace onnxruntime {
+/*
+This transform changes the following subgraph pattern:
+Cast --> IsInf --> Cast --> ReduceSum --> Greater
+to
+IsAllFinite --> Not
+*/
+Status IsInfReduceSumFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level, const logging::Logger& logger) const {
+  GraphViewer graph_viewer(graph);
+  const auto& node_topology_list = graph_viewer.GetNodesInTopologicalOrder();
+  std::vector<std::reference_wrapper<Node>> nodes_to_remove;
+  for (auto node_index : node_topology_list) {
+    nodes_to_remove.clear();
+    auto* node_ptr = graph.GetNode(node_index);
+    if (!node_ptr)
+      continue;  // node was removed
+
+    auto& isinf_node = *node_ptr;
+
```

Comment:
`ORT_ENFORCE(node_ptr->OpType() == "IsInf"` maybe?

Reply:
we check for it in the next few lines